crates/polars-plan/src/plans/ir/mod.rs

```diff
@@ -203,6 +210,844 @@ impl IRPlan {
     pub fn display_dot(&self) -> dot::IRDotDisplay<'_> {
         self.as_ref().display_dot()
     }
+
+    /// Convert an IR plan to a template by replacing all data sources with PlaceholderScan nodes.
+    ///
+    /// This traverses the IR tree and replaces:
+    /// - `IR::DataFrameScan` → `IR::PlaceholderScan`
+    /// - `IR::Scan` (CSV, Parquet, etc.) → `IR::PlaceholderScan`
+    /// - `IR::PythonScan` → `IR::PlaceholderScan`
+    ///
+    /// All other IR nodes (Filter, Select, Join, etc.) are preserved as-is.
+    ///
+    /// # Important: Only Use with Unoptimized Plans
+    ///
+    /// This method should only be called on **unoptimized** IR plans where:
```

Comment:
Do you have an idea of how this could be implemented?

It would solve these issues:
- https://github.com/pola-rs/polars/issues/19906
- https://github.com/pola-rs/polars/issues/22331